src/chem/viirs_aod2ioda.py

```diff
@@ -10,42 +11,74 @@
 import argparse
 import netCDF4 as nc
 import numpy as np
-from datetime import datetime, timedelta
+from datetime import datetime, date, timedelta
+import os
 from pathlib import Path
 
 IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
 if not IODA_CONV_PATH.is_dir():
     IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
 sys.path.append(str(IODA_CONV_PATH.resolve()))
 
-import ioda_conv_ncio as iconv
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
 from orddicts import DefaultOrderedDict
 
-vName = {
-    'A': "aerosol_optical_depth_4",
-}
-
 locationKeyList = [
     ("latitude", "float"),
     ("longitude", "float"),
     ("datetime", "string")
 ]
 
-AttrData = {}
 
+obsvars = {
+    'A': "aerosol_optical_depth_4",
+}
 
-class AOD(object):
+AttrData = {
+    'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
 
-    def __init__(self, filename, method, mask, thin, writer):
+
+DimDict = {
+}
+
+VarDims = {
+    'aerosol_optical_depth_4': ['nlocs']
+}
+
+
+class AOD(object):
+    def __init__(self, filename, method, mask, thin):
+        print('filename= ', filename)
+        print('pwd=', os.getcwd())
         self.filename = filename
-        self.method = method
+        print('self.filename= ', self.filename)
         self.mask = mask
+        self.method = method
         self.thin = thin
-        self.data = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
-        self.writer = writer
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
         self._read()
 
     def _read(self):
+        # set up variable names for IODA
+        for ncvar, iodavar in obsvars.items():
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'unitless'
+            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'unitless'
```

Comment:
After some discussion, we settled on "unitless" being the units name for categorical variables such as qc flags and enumerations. Whereas for physical quantities that happen to be without units, we settled on using '1' as the units name. We are doing this so we can distinguish between variables that can be used in mathematical calculations (units = "1") from those that cannot (units = "unitless").

I'm thinking that aerosol_optical_depth is a unit-less physical quantity that can be used in mathematical formulas. If that is correct, then please set the units to '1'.

Thanks!

Reply:
I've made this correction. But I'm having a ctest failure (also noted in the modis_aod2ioda PR):      

Start 446: test_iodaconv_ncep_bufr
1/1 Test #446: test_iodaconv_ncep_bufr ..........***Failed    0.38 sec
RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/da/Andrew.Tangborn/JEDI/viirs_codesprint/build/bin/ncep_classes.py", line 3, in <module>
    import ncepbufr
  File "/scratch1/NCEPDEV/jcsda/jedipara/opt/modules/intel-2020.2/bufr/noaa-emc-11.5.0/lib/python3.7/site-packages/ncepbufr/__init__.py", line 1, in <module>
    import _bufrlib
ImportError: numpy.core.multiarray failed to import

I'm not sure what to do here. Is there an updated version that fixes this?